### DISCUSSION TITLE
Weird behaviour in .NET Framework 4.8

### DISCUSSION THREAD

**Discussion by dmki** [betalgo/openai#254] [Q&A]
**Describe the bug**
When simple test code is executed in .NET Framework 4.8 project of type Class Library or Console - it works. When it's executed in Windows Form (e.g. in button click) - it doesn't, even if button is calling the DLL that is actually executing the code.

**Your code piece**
Consider this code:
```
	    string _apiKey = @""; //API key here
            var cr = new ChatCompletionCreateRequest();
            cr.MaxTokens = 25;
            cr.Temperature = 0;
            cr.TopP = 0;
            cr.PresencePenalty = 0;
            cr.FrequencyPenalty = 0;
            cr.Messages = new List<ChatMessage>();
            cr.Messages.Add(new ChatMessage("user", "2+2=?"));
            var _service = new OpenAIService(new OpenAiOptions()
            {
                ApiKey = _apiKey
                , DefaultModelId = "gpt-3.5-turbo"
            });
            var response = _service.ChatCompletion.CreateCompletion(cr).Result;
            if (response.Successful)
            {
                MessageBox.Show("2+2=" + response.Choices.First().Message);
            } else
            {
                MessageBox.Show("ERROR");
            }
```
**Result**
It returns "4" if your project type is Console and you are executing it in console. When it's executed from Form - the call is never returned, it will wait forever.

**Expected behavior**
I expected the code to work, or at least return an error explaining what's wrong.

I assume there is a simple explanation to this behaviour that I am missing, but perhaps DLL can handle it?

**Comment by luzemin** [comment]
This is a common mistake.
In UI applications, calling asynchronous methods using synchronous methods results in deadlocks.

Incorrect code line：
`ar response = _service.ChatCompletion.CreateCompletion(cr).Result;`

Please refer to the following article：
https://stackoverflow.com/questions/15021304/an-async-await-example-that-causes-a-deadlock
https://blog.stephencleary.com/2012/07/dont-block-on-async-code.html
